contribute/contribution-setup.md

```diff
@@ -0,0 +1,86 @@
+# OpenEBS Contributing Setup Guide
+
+Fork the [openebs/openebs](https://github.com/openebs/openebs) repository into your account, referred in the below instructions as $user. You can follow the same approach for all the other repositories where you want to contribute.
+
+## Setting up Development Environment for OpenEBS projects
+
+```bash
+working_dir=$GOPATH/src/github.com/openebs
+mkdir -p $working_dir
+cd $working_dir
+```
+
+Set `user` to match your Github profile name:
+for setting global configurations use - 
+git config --global user.name "your usernae here"
+git config --global user.email "your email used in github account"
+
+and for configuring locally use the above commands just remove the --global flag
+```bash
+user={your Github profile name}
+```
+
+## The steps mentioned here are w.r.t contribution towards openebs/openebs project
+
+```bash
+git clone https://github.com/$user/openebs.git
+cd openebs/
+git remote add upstream https://github.com/openebs/openebs.git
+git remote set-url --push upstream no_push
+git remote -v
+```
+
+## Synchronizing your local develop(i.e. your $user/openebs project) with upstream develop(i.e. openebs/openebs)
+
+```bash
+git checkout develop
+git fetch upstream develop
+git rebase upstream/develop
+git status
+git push origin develop
+```
+
+## Create Branch (based on feature or bug name)
+
+```bash
+git branch <branch_name>
+git checkout <branch_name>
+git push --set-upstream origin <branch_name>
+```
+
+## Synchronizing (i.e rebase) your local branch with upstream develop
+
+```bash
+git checkout <branch-name>
+git fetch upstream develop
+git rebase upstream/develop
+git status
+git push
+```
+
+## Make changes to your local branch
+
+```bash
+# make your changes
+# keep fetching the commits from upstream develop & rebase them here
+git add -A
+git commit -m "creating changes in local repository"
+git push
+
+# submit the PR to upstream from browser link https://github.com/$user/openebs
+```
+
+## Delete Branch
+
+```bash
+git push origin --delete <branch_name>
+git branch -d <branch_name>
```

Comment:
This requires a fully merged branch. Did you mean to use `-D`?

Reply:
I did not change anything, it was present on the openebs/openebs just moved it here. Want me to use -D?

Reply:
Sure... I was reviewing the changed lines, so poked at it. `-d` is fine I guess.